Building with QUIC_BUILD_SHARED=OFF not working?

```
lars@ubuntuvm /m/l/D/C/m/Linux (main)> cmake -GNinja -DQUIC_BUILD_SHARED=OFF ..
-- CMAKE Version: 3.16.3
-- Source Dir: /mnt/lars/Documents/Code/msquic
-- Host System name: Linux
-- Setting policy 0091
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- System name: Linux
-- System version: 5.4.0-86-generic
-- Platform version:
-- Build type: Release
-- QUIC Platform: linux
-- Version Build ID: 0
-- Version Suffix: -private
-- Looking for _SC_PHYS_PAGES
-- Looking for _SC_PHYS_PAGES - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for UDP_SEGMENT
-- Looking for UDP_SEGMENT - found
-- Looking for SO_ATTACH_REUSEPORT_CBPF
-- Looking for SO_ATTACH_REUSEPORT_CBPF - found
-- Looking for sendmmsg
-- Looking for sendmmsg - found
-- Enabling OpenSsl configuration tests
-- Enabling shared ephemeral port work around
-- Disabling tracing
-- Setting openssldir to /usr/lib/ssl
-- Found libatomic: /usr/lib/x86_64-linux-gnu/libatomic.so.1
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Configuring for OpenSSL
CMake Error at src/bin/CMakeLists.txt:78 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "OUTPUT_NAME" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:100 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_LIBRARIES" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:102 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_DEPENDS" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:78 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "OUTPUT_NAME" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:100 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_LIBRARIES" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:102 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_DEPENDS" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:78 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "OUTPUT_NAME" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:112 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:100 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_LIBRARIES" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:112 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:102 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_DEPENDS" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:112 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:78 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "OUTPUT_NAME" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:100 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_LIBRARIES" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


CMake Error at src/bin/CMakeLists.txt:102 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_DEPENDS" is not allowed.
Call Stack (most recent call first):
  src/bin/CMakeLists.txt:106 (flatten_link_dependencies)
  src/bin/CMakeLists.txt:117 (flatten_link_dependencies)


-- "/mnt/lars/Documents/Code/msquic/Linux/obj/$<IF:$<CONFIG:Debug>,Debug,Release>/libmsquic_static.a"
-- "/mnt/lars/Documents/Code/msquic/Linux/obj/$<IF:$<CONFIG:Debug>,Debug,Release>/libcore.a"
-- "/mnt/lars/Documents/Code/msquic/Linux/obj/$<IF:$<CONFIG:Debug>,Debug,Release>/libplatform.a"
-- "/mnt/lars/Documents/Code/msquic/Linux/_deps/opensslquic-build/openssl/lib/libssl.a"
-- "/mnt/lars/Documents/Code/msquic/Linux/_deps/opensslquic-build/openssl/lib/libcrypto.a"
CMake Error at src/bin/CMakeLists.txt:233 (set_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "FOLDER" is not allowed.


CMake Error at src/bin/CMakeLists.txt:247 (SET_TARGET_PROPERTIES):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "LINK_FLAGS" is not allowed.


-- Found Python: /usr/bin/python3.8 (found version "3.8.10") found components: Interpreter
-- Configuring incomplete, errors occurred!
See also "/mnt/lars/Documents/Code/msquic/Linux/CMakeFiles/CMakeOutput.log".
See also "/mnt/lars/Documents/Code/msquic/Linux/CMakeFiles/CMakeError.log".
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -137,6 +137,10 @@ option(QUIC_DISABLE_POSIX_GSO "Disable GSO for systems that say they support it 
 set(QUIC_FOLDER_PREFIX "" CACHE STRING "Optional prefix for source group folders when using an IDE generator")
 set(QUIC_LIBRARY_NAME "msquic" CACHE STRING "Override the output library name")
 
+if (NOT QUIC_BUILD_SHARED)
+    cmake_minimum_required(VERSION 3.20)
+endif()
+
 set(BUILD_SHARED_LIBS ${QUIC_BUILD_SHARED})
 
 # FindLTTngUST does not exist before CMake 3.6, so disable logging for older cmake versions

--- a/src/bin/CMakeLists.txt
+++ b/src/bin/CMakeLists.txt
@@ -133,18 +133,12 @@ else()
             OUTPUT ${QUIC_DEPS_FILE}
             CONTENT "${DEPS_STRING}"
         )
-        add_custom_target(
-            msquic_linker_inputs
-            DEPENDS ${QUIC_DEPS_FILE}
-        )
-        add_dependencies(msquic_linker_inputs msquic_static)
-        set_property(TARGET msquic_linker_inputs PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}libraries")
 
         # Run archiver tool
-        add_custom_target(
-            msquic_lib
+        add_custom_command(
+            OUTPUT ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
             COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-            COMMAND libtool -static -o \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libmsquic.a\" @${QUIC_DEPS_FILE}
+            COMMAND libtool -static -o \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}\" @${QUIC_DEPS_FILE}
             DEPENDS ${QUIC_DEPS_FILE}
         )
     elseif(WIN32)
@@ -164,21 +158,15 @@ else()
         file(
             GENERATE
             OUTPUT ${QUIC_DEPS_FILE}
-            CONTENT "/OUT:\"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/msquic.lib\" ${DEPS_STRING} ${NODEFAULTS}"
+            CONTENT "/OUT:\"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}\" ${DEPS_STRING} ${NODEFAULTS}"
         )
-        add_custom_target(
-            msquic_linker_inputs
-            DEPENDS ${QUIC_DEPS_FILE}
-        )
-        add_dependencies(msquic_linker_inputs msquic_static)
-        set_property(TARGET msquic_linker_inputs PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}libraries")
 
         # Run archiver tool (lib.exe) on our generated linker args
         add_custom_target(
-            msquic_lib
+            OUTPUT ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
             COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
             COMMAND ${CMAKE_AR} @${QUIC_DEPS_FILE}
-            COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/msquic.lib" "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}"
+            COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}" "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}"
             DEPENDS ${QUIC_DEPS_FILE}
         )
     else()
@@ -194,22 +182,23 @@ else()
             OUTPUT ${QUIC_DEPS_FILE}
             CONTENT "create \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libmsquic.a\"\n${DEPS_LINES}save\nend\n"
         )
-        add_custom_target(
-            msquic_linker_inputs
-            DEPENDS ${QUIC_DEPS_FILE}
-        )
-        add_dependencies(msquic_linker_inputs msquic_static)
-        set_property(TARGET msquic_linker_inputs PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}libraries")
 
         # Run archiver tool
-        add_custom_target(
-            msquic_lib
+        add_custom_command(
+            OUTPUT ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
             COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
             COMMAND ${CMAKE_AR} -M < ${QUIC_DEPS_FILE}
             DEPENDS ${QUIC_DEPS_FILE}
+            DEPENDS msquic_static
         )
     endif()
-    add_dependencies(msquic_lib msquic_linker_inputs)
+
+    add_custom_target(
+        msquic_lib
+        DEPENDS
+        ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
+    )
+
     set_property(TARGET msquic_lib PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}libraries")
 
     # Provide interface library to link the monolithic library
@@ -219,11 +208,12 @@ else()
     # msquic_static directly if they don't need the monolithic library.
     add_library(msquic INTERFACE)
     target_compile_definitions(msquic INTERFACE QUIC_BUILD_STATIC)
+    target_link_libraries(msquic INTERFACE inc)
     target_link_libraries(msquic
         INTERFACE
         ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
-        inc
     )
+    target_link_libraries(msquic INTERFACE inc)
     if (CX_PLATFORM STREQUAL "darwin")
         target_link_libraries(msquic INTERFACE "-framework CoreFoundation" "-framework Security")
     endif()

--- a/src/bin/CMakeLists.txt
+++ b/src/bin/CMakeLists.txt
@@ -162,7 +162,7 @@ else()
         )
 
         # Run archiver tool (lib.exe) on our generated linker args
-        add_custom_target(
+        add_custom_command(
             OUTPUT ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
             COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
             COMMAND ${CMAKE_AR} @${QUIC_DEPS_FILE}
@@ -200,6 +200,8 @@ else()
     )
 
     set_property(TARGET msquic_lib PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}libraries")
+
+    message(STATUS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}")
 
     # Provide interface library to link the monolithic library
     # NOTE: It is important that this interface library NOT link directly to

--- a/src/bin/CMakeLists.txt
+++ b/src/bin/CMakeLists.txt
@@ -11,7 +11,7 @@ endif()
 
 if(BUILD_SHARED_LIBS)
     add_library(msquic SHARED ${SOURCES})
-    target_link_libraries(msquic PRIVATE core platform inc warnings logging)
+    target_link_libraries(msquic PRIVATE core platform inc warnings logging base_link)
     set_target_properties(msquic PROPERTIES OUTPUT_NAME ${QUIC_LIBRARY_NAME})
 else()
     add_library(msquic_static STATIC static/empty.c)
@@ -174,13 +174,13 @@ else()
         set(QUIC_DEPS_FILE ${CMAKE_CURRENT_BINARY_DIR}/$<IF:$<CONFIG:Debug>,Debug,Release>/quicdeps.txt)
         set(DEPS_LINES "")
         foreach(line ${DEPS_LIST})
-            message(STATUS ${line})
+            # message(STATUS ${line})
             set(DEPS_LINES "${DEPS_LINES}addlib ${line}\n")
         endforeach(line)
         file(
             GENERATE
             OUTPUT ${QUIC_DEPS_FILE}
-            CONTENT "create \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libmsquic.a\"\n${DEPS_LINES}save\nend\n"
+            CONTENT "create \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}\"\n${DEPS_LINES}save\nend\n"
         )
 
         # Run archiver tool
@@ -208,12 +208,10 @@ else()
     # msquic_static directly if they don't need the monolithic library.
     add_library(msquic INTERFACE)
     target_compile_definitions(msquic INTERFACE QUIC_BUILD_STATIC)
-    target_link_libraries(msquic INTERFACE inc)
     target_link_libraries(msquic
         INTERFACE
         ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}
     )
-    target_link_libraries(msquic INTERFACE inc)
     if (CX_PLATFORM STREQUAL "darwin")
         target_link_libraries(msquic INTERFACE "-framework CoreFoundation" "-framework Security")
     endif()

--- a/src/bin/CMakeLists.txt
+++ b/src/bin/CMakeLists.txt
@@ -140,6 +140,7 @@ else()
             COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
             COMMAND libtool -static -o \"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}\" @${QUIC_DEPS_FILE}
             DEPENDS ${QUIC_DEPS_FILE}
+            DEPENDS msquic_static
         )
     elseif(WIN32)
         set(NODEFAULTS "")
@@ -168,6 +169,7 @@ else()
             COMMAND ${CMAKE_AR} @${QUIC_DEPS_FILE}
             COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${QUIC_LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}" "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}"
             DEPENDS ${QUIC_DEPS_FILE}
+            DEPENDS msquic_static
         )
     else()
         # Emit all linker inputs at build time to resolve generator expressions

--- a/src/core/unittest/CMakeLists.txt
+++ b/src/core/unittest/CMakeLists.txt
@@ -26,7 +26,7 @@ if (BUILD_SHARED_LIBS)
     target_link_libraries(msquiccoretest core platform)
 endif()
 
-target_link_libraries(msquiccoretest inc gtest warnings logging)
+target_link_libraries(msquiccoretest inc gtest warnings logging base_link)
 
 add_test(NAME msquiccoretest
          COMMAND msquiccoretest

--- a/src/inc/CMakeLists.txt
+++ b/src/inc/CMakeLists.txt
@@ -14,12 +14,14 @@ target_include_directories(inc INTERFACE ${QUIC_INCLUDE_DIR})
 target_compile_features(inc INTERFACE cxx_std_17)
 target_compile_features(inc INTERFACE c_std_11)
 
+add_library(base_link INTERFACE)
+
 if (HAS_GUARDCF)
-    target_link_options(inc INTERFACE /guard:cf /DYNAMICBASE)
+    target_link_options(base_link INTERFACE /guard:cf /DYNAMICBASE)
 endif()
 
 if (NOT MSVC AND QUIC_ENABLE_SANITIZERS)
-    target_link_libraries(inc INTERFACE -fsanitize=address,leak,undefined)
+    target_link_libraries(base_link INTERFACE -fsanitize=address,leak,undefined)
 endif()
 
 if (NOT MSVC AND NOT APPLE AND NOT ANDROID)
@@ -33,14 +35,14 @@ endif()
 
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
-target_link_libraries(inc INTERFACE Threads::Threads ${CMAKE_DL_LIBS})
+target_link_libraries(base_link INTERFACE Threads::Threads ${CMAKE_DL_LIBS})
 
 if (ATOMIC)
-    target_link_libraries(inc INTERFACE ${ATOMIC})
+    target_link_libraries(base_link INTERFACE ${ATOMIC})
 endif()
 
 if(WIN32)
-    target_link_libraries(inc INTERFACE ws2_32 schannel ntdll bcrypt ncrypt crypt32 iphlpapi advapi32)
+    target_link_libraries(base_link INTERFACE ws2_32 schannel ntdll bcrypt ncrypt crypt32 iphlpapi advapi32)
 endif()
 
 add_library(warnings INTERFACE)

--- a/src/perf/bin/CMakeLists.txt
+++ b/src/perf/bin/CMakeLists.txt
@@ -15,4 +15,4 @@ if (BUILD_SHARED_LIBS)
     target_link_libraries(secnetperf platform)
 endif()
 
-target_link_libraries(secnetperf logging)
+target_link_libraries(secnetperf logging base_link)

--- a/src/platform/unittest/CMakeLists.txt
+++ b/src/platform/unittest/CMakeLists.txt
@@ -22,7 +22,7 @@ if (BUILD_SHARED_LIBS)
     target_link_libraries(msquicplatformtest platform)
 endif()
 
-target_link_libraries(msquicplatformtest inc gtest warnings logging)
+target_link_libraries(msquicplatformtest inc gtest warnings logging base_link)
 
 add_test(NAME msquicplatformtest
          COMMAND msquicplatformtest

--- a/src/test/bin/CMakeLists.txt
+++ b/src/test/bin/CMakeLists.txt
@@ -18,7 +18,7 @@ if (BUILD_SHARED_LIBS)
     target_link_libraries(msquictest platform)
 endif()
 
-target_link_libraries(msquictest inc gtest logging)
+target_link_libraries(msquictest inc gtest logging base_link)
 
 # At least /W3 must be used on all windows builds to pass compliance
 if(MSVC)

--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -6,13 +6,13 @@ function(add_quic_tool)
     list(REMOVE_AT ARGV 0)
 
     add_executable(${targetname} ${ARGV})
-    target_link_libraries(${targetname} inc warnings msquic inc logging)
+    target_link_libraries(${targetname} inc warnings msquic)
 
     if (BUILD_SHARED_LIBS)
         target_link_libraries(${targetname} platform)
     endif()
 
-    target_link_libraries(${targetname} base_link)
+    target_link_libraries(${targetname} logging base_link)
 
     set_property(TARGET ${targetname} PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}tools")
 endfunction()

--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -12,7 +12,7 @@ function(add_quic_tool)
         target_link_libraries(${targetname} platform)
     endif()
 
-    target_link_libraries(${targetname} inc logging)
+    target_link_libraries(${targetname} base_link)
 
     set_property(TARGET ${targetname} PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}tools")
 endfunction()

